code/__DATASTRUCTURES/stacks.dm

```diff
@@ -0,0 +1,55 @@
+
+//Both count as failures, and they don't equate to each other
+//this lets us do if(pop()) without having to specifically check for underflow
+//same for if(push()) and overflow
+#define STACK_OVERFLOW	""
```

Comment:
I think it would be clearer if you made the DEFINEs into negative numbers. They would still count as false and it wouldn't be as confusing.